c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -3338,7 +3353,11 @@ class DeviceCachingAllocator {
         EventPool::Event event = std::move(e.first);
         Block* block = e.second;
 
-        cudaError_t err = C10_CUDA_ERROR_HANDLED(cudaEventQuery(*event));
+        cudaError_t err{};
+        {
+          at::cuda::CUDAStreamCaptureModeGuard g{cudaStreamCaptureModeRelaxed};
```

Comment:
If this is needed to do the `cudaEventQuery`, it might hurt debuggability a bit if it means user-disallowed code can slip through here. However this is probably unlikely as I would assume the Python thread would not be making progress while we are here